src/index.ts

```diff
@@ -1,36 +1,34 @@
 let vacias: number = 0;
 let ocupadas: number = 0;
-let indice: number;
-
 let cantButacas: number;
 cantButacas = Number(prompt("Ingrese la cantidad de butacas")); //pide al usuario ingresar cantidades de butacas
 
-let butacas: number[] = new Array(cantButacas); //defino el array
+let butacas: boolean[] = new Array(cantButacas); //defino el array
 
 //----------------------------------------------
 //funcion para cargar el arreglo con booleanos
-function cargarArrayRandomBoolean(v: number, cantButacas: number) {
+function cargarArrayRandomBoolean(vector: boolean[], cantButacas: number) {
   for (let indice = 0; indice < cantButacas; indice++) {
-    v[indice] = Boolean(Math.floor(Math.random() * 2));
+    vector[indice] = Boolean(Math.floor(Math.random() * 2));
   }
-  return v;
+  return vector;
 }
 //-------------------------------------
 
-console.log(cargarArrayRandomBoolean(butacas, cantButacas));//muestra el arreglo cargado
+console.log(cargarArrayRandomBoolean(butacas, cantButacas)); //muestra el arreglo cargado
 
 //-------------------------------------------
 //funcion que verifica las cantidades de asientos ocupados y vacios
-function verificarButacas (cargarArrayRandomBoolean():number){
-for (indice = 0; indice < cantButacas; indice++) {
-  if (butacas[indice] == 1) {
-    ocupadas += 1;
-  } else {
-    vacias += 1;
+function verificarButacas(vector: boolean[]) {
+  for (let indice = 0; indice < cantButacas; indice++) {
+    if (vector[indice] === true) {
+      ocupadas += 1;
+    } else {
+      vacias += 1;
+    }
   }
 }
```

Comment:
La función **verificarButacas** esta mal.
- Porque pasamos la función cargarArrayRandomBoolean() como parámetro de la función **verificarButacas()** ?
- Además la funcion **cargarArrayRandomBoolean** retorna un arreglo booleano, no un número. Porque pusiste `:number`?

Lo correcto seria que la función **verificarButacas** reciba como parámetro un array booleano `function verificarButacas(vector: boolean[])`.
Después dentro de la función esta bien el for, solo que en el if comparamos el valor con **true** y no con **1**, porque el arreglo es de booleanos.